### PR TITLE
Switch image_tag to tag(img) for external icon URLs

### PR DESCRIPTION
**Pull request by Throne3d**

### CODE REVIEW

File: app/views/layouts/application.haml [hunk 1]
```diff
@@ -60,7 +60,7 @@
           #user-info
             = link_to current_user do
               - if current_user.avatar
-                = image_tag current_user.avatar.url, class: 'icon', id: 'user-icon', title: current_user.avatar.keyword, alt: current_user.avatar.keyword
+                %img{src: current_user.avatar.url, class: 'icon', id: 'user-icon', title: current_user.avatar.keyword, alt: current_user.avatar.keyword}
```
[the remark this answers is not in the document]
Comment (Throne3d) [reply app/views/layouts/application.haml:63]:
```suggestion
                %img#user-icon.icon{src: current_user.avatar.url, title: current_user.avatar.keyword, alt: current_user.avatar.keyword}
```

File: app/helpers/icon_helper.rb [hunk 2]
```diff
@@ -19,7 +19,7 @@ def icon_mem_tag(url, keyword, **args)
       klass += ' ' + supplied_class
     end
 
-    image_tag url, { alt: keyword, title: keyword, class: klass }.merge(**args)
+    tag.img({ src: url, alt: keyword, title: keyword, class: klass }.merge(**args))
```
Comment (Throne3d) [inline app/helpers/icon_helper.rb:22]:
```suggestion
    tag.img(src: url, alt: keyword, title: keyword, class: klass, **args)
```

File: app/helpers/icon_helper.rb [hunk 3]
```diff
@@ -28,7 +28,7 @@ def no_icon_tag(**args)
 
   def quick_switch_tag(image_url, short_text, hover_name, char_id)
     return tag.div short_text, class: CHAR_ICON_FAKE, title: hover_name, data: { character_id: char_id } if image_url.nil?
-    image_tag image_url, class: CHAR_ICON, alt: hover_name, title: hover_name, data: { character_id: char_id }
+    tag.img({ src: image_url, class: CHAR_ICON, alt: hover_name, title: hover_name, data: { character_id: char_id } })
```
Comment (Throne3d) [inline app/helpers/icon_helper.rb:31]:
```suggestion
    tag.img(src: image_url, class: CHAR_ICON, alt: hover_name, title: hover_name, data: { character_id: char_id })
```

File: spec/support/spec_test_helper.rb [hunk 4]
```diff
@@ -46,3 +46,32 @@ def stub_fixture(url, filename)
...
+    else
+      { tag: n.node_name, path: n.path, content: n.content }
+    end
+  end
+end
+
+RSpec::Matchers.define :eq_structured_html do |expected|
+  match do |actual|
+    @actual = structured_html(actual).sort_by(&:to_s)
+    @expected = expected.sort_by(&:to_s)
```
Comment (teceler) [inline spec/support/spec_test_helper.rb:71]: You might want something here to turn the expected into structured html if it's a string?

Comment (Throne3d) [reply spec/support/spec_test_helper.rb:71]: Ooh, that's a good idea!

Comment (Throne3d) [inline spec/support/spec_test_helper.rb:71]:
```suggestion
    @expected = if expected.is_a?(String)
      structured_html(expected)
    else
      expected
    end.sort_by(&:to_s)
```

**Pull request merged**